### PR TITLE
[Google Blockly] check for the presence of blocks

### PR DESCRIPTION
**Pull request by mikeharv**
@fisher-alice noticed an issue with the new [xml->json conversion](https://github.com/code-dot-org/code-dot-org/pull/52072). There was an uncaught error if there were no blocks in the serialized JSON state. https://codedotorg.slack.com/archives/C03DBDN67B7/p1686238446034199?thread_ts=1686001162.405249&cid=C03DBDN67B7
Screen Shot 2023-06-08 at 10 33 37 AM

I observed that in each case where the error was happening, that `xmlBlocks` was an empty array. When the JSON serialization system saves a blank workspace, the result is just `{}`, so we cannot assume that the `blocks` property will be present.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/addons/cdoSerializationHelpers.js [hunk 1]
```diff
@@ -25,13 +25,15 @@ export function convertXmlToJson(xml) {
...
   const stateToLoad = Blockly.serialization.workspaces.save(tempWorkspace);
 
-  // Create a map of ids (key) and block serializations (value).
-  const blockIdMap = stateToLoad.blocks.blocks.reduce(
-    (map, blockJson) => map.set(blockJson.id, blockJson),
-    new Map()
-  );
-
-  addPositionsToState(xmlBlocks, blockIdMap);
+  if (xmlBlocks.length && stateToLoad.blocks) {
```
Comment (fisher-alice) [inline apps/src/blockly/addons/cdoSerializationHelpers.js:28]: Just curious if there's ever a case when one condition is `true` and the other is `false`.

Comment (fisher-alice) [inline apps/src/blockly/addons/cdoSerializationHelpers.js:28]: Just curious if there's ever a case when one condition is `true` and the other is `false`.

**Review by fisher-alice — APPROVED** [review]
Thanks for taking care of this so quickly - just a small question.

File: apps/src/blockly/addons/cdoSerializationHelpers.js (hunk 1 above)
Comment (mikeharv) [reply apps/src/blockly/addons/cdoSerializationHelpers.js:28]: I don't see how they could be different, but it felt like it was clearer to communicate that we're expecting both.

**Pull request merged**